mirdata/datasets/phenicx_anechoic.py

```diff
@@ -0,0 +1,421 @@
+  journal={Journal of Electrical and Computer Engineering},
+  volume={2016},
+  year={2016},
+  publisher={Hindawi}
+}
+@article{patynen2008anechoic,
+  title={Anechoic recording system for symphony orchestra},
+  author={P{\"a}tynen, Jukka and Pulkki, Ville and Lokki, Tapio},
+  journal={Acta Acustica united with Acustica},
+  volume={94},
+  number={6},
+  pages={856--865},
+  year={2008},
+  publisher={S. Hirzel Verlag}
+}
+"""
+
+REMOTES = {
+    "all": download_utils.RemoteFileMetadata(
+        filename="PHENICX-Anechoic.zip",
+        url="https://zenodo.org/record/840025/files/PHENICX-Anechoic.zip?download=1",
+        checksum="7fec47568263476ecac0103aef608629",
+        #destination_dir="..",
+        unpack_directories=["PHENICX-Anechoic"]
+    )
+}
+
+LICENSE_INFO = """
+Creative Commons Attribution Non Commercial Share Alike 4.0 International
+"""
+
+DATASET_SECTIONS = {
+    "doublebass": "strings",
+    "cello": "strings",
+    "clarinet": "woodwinds",
+    "viola": "strings",
+    "violin": "strings",
+    "oboe": "woodwinds",
+    "flute": "woodwinds",
+    "trumpet": "brass",
+    "bassoon": "woodwinds",
+    "horn": "brass",
+}
+
+
+class Track(core.Track):
+    """Phenicx-Anechoic Track class
+
+    Args:
+        track_id (str): track id of the track
+
+    Attributes:
+        audio_path (list): path to the audio files
+        notes_path (list): path to the audio files
+        notes_original_path (list): path to the audio files
+        instrument (str): the name of the instrument
+        piece (str): the name of the piece
+        n_voices (int): the number of voices in this instrument
+        track_id (str): track id
+
+    Cached Properties:
+        melody (F0Data): melody annotation
+
+    """
+
+    def __init__(
+        self,
+        track_id,
+        data_home,
+        dataset_name,
+        index,
+        metadata,
+    ):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+
+        self.instrument = self.track_id.split("-")[1]
+        self.piece = self.track_id.split("-")[0]
+
+        self.audio_paths = [
+            os.path.join(self._data_home, self._track_paths[key][0])
+            for key in self._track_paths
+            if "audio_" in key
+        ]
+
+        self.n_voices = len(self.audio_paths)
+
+        self.notes_path = os.path.join(self._data_home, self._track_paths["notes"][0])
+
+        self.notes_original_path = os.path.join(
+            self._data_home, self._track_paths["notes_original"][0]
+        )
+
+    @property
+    def audio(self) -> Optional[Tuple[np.ndarray, float]]:
+        """the track's audio
+
+        Returns:
+            * np.ndarray - the mono audio signal
+            * float - The sample rate of the audio file
+
+        """
+        assert self.audio_paths is not None and isinstance(self.audio_paths,list)
```

Comment:
from the init, audio_paths will never be None, but it could be an empty list. @drubinstein suggested that this assert could be what's causing mypy to flag it. Try removing the assert!

Reply:
A bit of a hack, but you could replace the load audio lines with
```
        audio_mix, sr = cast(Tuple[np.ndarray, float], load_audio(self.audio_paths[0]))

        for i in range(1, self.n_voices):
            audio, _ = cast(Tuple[np.ndarray, float], load_audio(self.audio_paths[i]))
      ```
      
      Alternatively (and maybe more correct), you could use a temporary variable. Something like 
      ```
      x = load_audio(self.audio_paths[i]
      audio, _ = x
      ```
      
      Going to point out that this code will break if one audio path cannot be opened or is None as you can't add None to an np.ndarray (nor can you add None to None)

Reply:
thanks @drubinstein ! the first solution worked while the second hasn't. so I have done that